src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,64 @@ export class CodeActionModel extends Disposable {
 					return;
 				}
 
-				const actions = createCancelablePromise(token => getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._showInlineQuickfxies()) {
+						const codeActionSet = await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
+
+						// Search for quickfixes in the curret code action set.
+						const foundQuickfix = codeActionSet.validActions?.some(action => action.action.kind === 'quickfix') || false;
+
+						const allMarkers = this._markerService.read({ resource: model.uri });
+
+						// If markers exists, and there are no quickfixes found or length is zero, check for quickfixes on that line.
+						if (allMarkers.length > 0 && (!foundQuickfix || codeActionSet.validActions.length === 0) && trigger.trigger.type === CodeActionTriggerType.Invoke) {
+							const currPosition: Position = trigger.selection.getPosition();
+							let trackedPosition: Position = currPosition;
+							let distance = Number.MAX_VALUE;
+							let toBeModified = false;
+
+							allMarkers.forEach((marker: IMarker) => {
+								const col = marker.endColumn;
+								const row = marker.endLineNumber;
+
+								// Found quickfix on the same line and check relative distance to other markers
+								if (row === currPosition.lineNumber && Math.abs(currPosition.column - col) < distance) {
+									distance = Math.abs(currPosition.column - col);
+									toBeModified = true;
+									trackedPosition = new Position(row, col);
+								}
+							});
+
+							// Only retriggers if actually found quickfix on the same line as cursor
+							if (toBeModified) {
+								const actionsAtMarker = await getCodeActions(this._registry, model, trigger.selection.toPositions(trackedPosition), trigger.trigger, Progress.None, token);
```

Comment:
When you re-request the code actions here, make sure of two tings:

1. `CodeActionFilter.include` in the call should be used so that you only request quick fixes
2. The caller should get a `CodeActionContext` with just the diagnostic you are interested in. Right now the diagnostic is populated on the extension host side which may not always find the correct diagnostic